### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by memoryleak47**

### CODE REVIEW

File: lang/step.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ For now, that is just a pointer (but this might have to change).
 Place evaluation ensures that this pointer is always dereferenceable (for the type of the place expression).
 
 ```rust
-type Place<M: Memory> = Pointer<M::Provenance>;
```
Comment (memoryleak47) [inline lang/step.md:135]: Rusts type aliases don't support bounds as of now.
We could implement something in specr, but I think this change is fine too.

File: lang/types.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ In the future, we might want to separate a type from its layout, and consider th
 
 ```rust
 /// A "layout" describes the shape of data in memory.
-struct Layout {
+pub struct Layout {
```
Comment (memoryleak47) [inline lang/types.md:24]: some modules from mem use `Layout`

File: lang/well-formed.md [hunk 3]
```diff
@@ -186,7 +187,10 @@ impl ValueExpr {
                         Type::Int(IntType { signed: Unsigned, size: M::PTR_SIZE })
                     }
                     Int2Ptr(ptr_ty) => {
-                        ensure(matches!(operand, Type::Int(IntType { signed: Unsigned, size: M::PTR_SIZE })))?;
```
Comment (memoryleak47) [inline lang/well-formed.md:189]: Sadly one cannot match like this `size: M::PTR_SIZE`.

File: lang/step.md (hunk 1 above)
Comment (RalfJung) [reply lang/step.md:135]: This will probably make the Coq translator unhappy, but yeah I guess it works for now.

File: lang/well-formed.md (hunk 3 above)
Comment (RalfJung) [reply lang/well-formed.md:189]: Does `Type` have equality? Then we could just use `==` instead of `matches!`?

Comment (memoryleak47) [reply lang/well-formed.md:189]: Yep, thats better!

**Pull request merged**